canon.go

```diff
@@ -91,3 +91,38 @@ func canonValue(chain *chain, in interface{}) (interface{}, bool) {
 
 	return out, true
 }
+
+func canonDecode(chain *chain, value interface{}, target interface{}) {
+	if chain.failed() {
+		return
+	}
+
+	if target == nil {
+		chain.fail(AssertionFailure{
+			Type:   AssertNil,
+			Actual: &AssertionValue{target},
+			Errors: []error{
+				errors.New("expected: target should not be nil"),
```

Comment:
Let's reword this: `expected: target should not be nil` -> `unexpected nil target argument` (we use same wording in other places)